data_gen/lagrangebench_data/count_files.py

```diff
@@ -0,0 +1,29 @@
+"""Count the number of files in each subdirectory of a direcory. For sanity check."""
```

Comment:
You don't need a python program for this. It can be done with a bash command, something like
```bash
find <DIR> -type f | sed 's%/[^/]*$%%'| uniq -c | awk '{if ($1 != <TARGET_COUNT>) $1="Failed!"; print}'
```

Reply:
I'm sure one could do sth like that. But I never bothered. What your current one-lines lacks is printing the actual number of files per directory. I used `count_files.py` to find out which dam break simulations blow up and need to be removed. For dam break, I had to run 115 instead of 100 simulations because of numerical instabilities.

If you want to remove the Python file, please add the full prints.

Reply:
What do you mean? It is printing the file count on the left of the dir name, and replaces it with "failed" if it differs from the target. This is what the output looks like:
```
Failed! /tmp/lagrangebench_data/raw/2D_TGV_2500_10kevery100/2D_TGV_SPH_2_20240625-161305
    127 /tmp/lagrangebench_data/raw/2D_TGV_2500_10kevery100/2D_TGV_SPH_0_20240625-161149
    127 /tmp/lagrangebench_data/raw/2D_TGV_2500_10kevery100/2D_TGV_SPH_1_20240625-161228
```

If you want the count for all dirs add it to awk as such
```bash
find<DIR> -type f | sed 's%/[^/]*$%%'| uniq -c | awk '{if ($1 != <TARGET_COUNT>) $1=$1" - Failed!"; n=split($2, a, /[/]/); print a[n]" -- "$1}'
```
which outputs
```
2D_TGV_SPH_2_20240625-161305 -- 112 - Failed!
2D_TGV_SPH_0_20240625-161149 -- 127
2D_TGV_SPH_1_20240625-161228 -- 127
```

Reply:
ok, I'm fine with the last version!

Reply:
Looks good